packages/@okta/vuepress-site/docs/guides/custom-widget-gen3/main/index.md

```diff
@@ -77,6 +77,100 @@ See [Customization examples](#customization-examples) for snippets that you can
 
    > **Note:** To discard your changes without publishing them, click **Revert changes** or click the **Code editor** toggle again. Turning off the code editor restores the default JavaScript code.
 
+### Use afterRender with the third generation
+
+The third generation of the Sign-In Widget is built on [Preact](https://preactjs.com/), a lightweight React alternative. This means that the `afterRender` function doesn't work when used to make DOM manipulations in the third generation.
+
+With the second generation of the Sign-In Widget, you can use [`afterRender`] (https://github.com/okta/okta-signin-widget?tab=readme-ov-file#afterrender)to change label text or input labels on a page.
+
+With the third generation, `afterRender` is incompatible with customizations that make DOM manipulations and other render-related side-effects (see [Components and Hooks must be pure](https://react.dev/reference/rules/components-and-hooks-must-be-pure)).
+
+#### Resolve the afterRender revert
+
+To keep the third generation of the Sign-In Widget from reverting your `afterRender` customizations, use the DOM `MutationObserver()` function. See [MutationObserver](https://dom.spec.whatwg.org/#ref-for-dom-mutationobserver-mutationobserver).
```

Comment:
Changed the reference link URL. Feel free to keep the original if providing the ref is standard. The MDN link is more useful.

```suggestion
To keep the third generation of the Sign-In Widget from reverting your `afterRender` customizations, use the `MutationObserver()` interface. See [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver).
```

Reply:
Providing the ref is standard.